File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        List<List<TreeNode>> depth = new ArrayList<>();
        if(root == null) {
            return depth.size();
        }

        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);

        while(!queue.isEmpty()) {
            int size = queue.size();
            List<TreeNode> list = new ArrayList<>();
            for(int i=0;i<size;i++) {
                TreeNode node = queue.poll();
                if(node.left != null) {
                    list.add(node.left);
                    queue.add(node.left);
                }
                if(node.right != null) {
                    list.add(node.right);
                    queue.add(node.right);
                }
            }
            depth.add(list);
            
        }

        return depth.size();
    }
}

File: Data Structures & Algorithms/path-sum/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:

        total_sum = 0 
        return self.helper(root, targetSum)


    def helper(self, root, total_sum):
        if not root:
            return False

        total_sum = total_sum - root.val

        if not root.left and not root.right:
            return total_sum == 0

        return self.helper(root.left, total_sum) or self.helper(root.right, total_sum)